newrelic/resource_newrelic_workload.go

```diff
@@ -86,14 +193,58 @@ func resourceNewRelicWorkload() *schema.Resource {
 	}
 }
 
+func WorkloadRuleRollupInputSchemaElem() *schema.Resource {
+	s := WorkloadRollupInputSchemaElem()
+	return &schema.Resource{
+		Schema: s,
+	}
+}
+
+func WorkloadremainingEntitiesRuleSchemaElem() *schema.Resource {
+	s := WorkloadRollupInputSchemaElem()
+
+	s["group_by"] = &schema.Schema{
+		Type:         schema.TypeString,
+		Required:     true,
+		Description:  "The grouping to be applied to the remaining entities.",
+		ValidateFunc: validation.StringInSlice(listValidWorkloadGroupBy(), false),
+	}
+
+	return &schema.Resource{
+		Schema: s,
+	}
+}
+
+func WorkloadRollupInputSchemaElem() map[string]*schema.Schema {
+	return map[string]*schema.Schema{
+		"strategy": {
+			Type:         schema.TypeString,
+			Required:     true,
+			Description:  "The rollup strategy that is applied to a group of entities.",
+			ValidateFunc: validation.StringInSlice(listValidWorkloadStrategy(), false),
+		},
+		"threshold_type": {
+			Type:         schema.TypeString,
+			Optional:     true,
+			Description:  "Type of threshold defined for the rule. This is an optional field that only applies when strategy is WORST_STATUS_WINS. Use a threshold to roll up the worst status only after a certain amount of entities are not operational.",
+			ValidateFunc: validation.StringInSlice(listValidWorkloadRuleThresholdType(), false),
+		},
+		"threshold_value": {
+			Type:        schema.TypeInt,
+			Optional:    true,
+			Description: "Threshold value defined for the rule. This optional field is used in combination with thresholdType. If the threshold type is null, the threshold value will be ignored.",
+		},
+	}
+}
+
 func resourceNewRelicWorkloadCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	client := meta.(*ProviderConfig).NewClient
 	createInput := expandWorkloadCreateInput(d)
 	accountID := d.Get("account_id").(int)
```

Comment:
We should use `selectAccountID` function here so user doesn't have to give account ID on the resource. See example - https://github.com/newrelic/terraform-provider-newrelic/blob/main/newrelic/resource_newrelic_alert_channel.go#L281